node/test/service/src/lib.rs

```diff
@@ -203,11 +203,12 @@ pub fn run_test_node(
 ) -> PolkadotTestNode {
 	let config = node_config(storage_update_func, task_executor, key, boot_nodes);
 	let multiaddr = config.network.listen_addresses[0].clone();
-	let authority_discovery_disabled = true;
+	let authority_discovery_disabled = false;
 	let NewFull {task_manager, client, network, rpc_handlers, overseer_handler, ..} =
 		polkadot_test_new_full(config, authority_discovery_disabled)
 			.expect("could not create Polkadot test service");
 
+	let overseer_handler = overseer_handler.expect("test node must have an overseer handler");
```

Comment:
This is not true. When I only start a full test node, it will not have the overseer handler.

Reply:
It's true since https://github.com/paritytech/polkadot/commit/c6e78be1de6942a6d8e9a63362265e5edb023b14.

I'll freely admit that I don't have a great understanding of the grand-scheme implications of enabling authority discovery for tests, but if this assertion were in fact false, then the tests would still be failing.